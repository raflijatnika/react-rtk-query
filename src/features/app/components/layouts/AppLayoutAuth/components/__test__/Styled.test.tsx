// Test
import rendered from 'react-test-renderer'
import 'jest-styled-components'

// Components
import {
  StyledWrapper,
  StyledContainer,
  StyledFormContainer,
  StyledBanner
} from '../Styled'

it('StyleWrapper should be rendered', () => {
  const tree = rendered.create(<StyledWrapper />).toJSON()

  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('display', 'flex')
  expect(tree).toHaveStyleRule('align-items', 'center')
  expect(tree).toHaveStyleRule('height', '100vh')
  expect(tree).toHaveStyleRule('width', '100vw')
})

it('StyledContainer should be rendered', () => {
  const tree = rendered.create(<StyledContainer />).toJSON()

  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('display', 'flex')
  expect(tree).toHaveStyleRule('align-items', 'center')
  expect(tree).toHaveStyleRule('justify-content', 'center')
  expect(tree).toHaveStyleRule('height', '100%')
  expect(tree).toHaveStyleRule('width', '100%')
})

it('StyledFormContainer should be rendered', () => {
  const tree = rendered.create(<StyledFormContainer />).toJSON()

  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('width', '30vw')
  expect(tree).toHaveStyleRule('height', '100%')
  expect(tree).toHaveStyleRule('background-color', '#fff')
  expect(tree).toHaveStyleRule('display', 'flex')
  expect(tree).toHaveStyleRule('justify-content', 'center')
  expect(tree).toHaveStyleRule('flex-direction', 'column')
  expect(tree).toHaveStyleRule('padding', '0px 68px')
  expect(tree).toHaveStyleRule('width', '100vw', {
    media: '(max-width: 1024px)'
  })
  expect(tree).toHaveStyleRule('padding', '0px 30px', {
    media: '(max-width: 1024px)'
  })
})

it('StyledBanner should be rendered', () => {
  const tree = rendered.create(<StyledBanner />).toJSON()

  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('width', '70vw')
  expect(tree).toHaveStyleRule('height', '100%')
  expect(tree).toHaveStyleRule('background-color', '#f1f4fa')
  expect(tree).toHaveStyleRule('display', 'flex')
  expect(tree).toHaveStyleRule('align-items', 'center')
  expect(tree).toHaveStyleRule('justify-content', 'center')
  expect(tree).toHaveStyleRule('display', 'none', {
    media: '(max-width: 1024px)'
  })
})
