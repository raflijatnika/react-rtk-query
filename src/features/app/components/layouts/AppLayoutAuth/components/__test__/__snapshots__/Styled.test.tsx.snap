// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleWrapper should be rendered 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

<div
  className="c0"
/>
`;

exports[`StyledBanner should be rendered 1`] = `
.c0 {
  background-color: #f1f4fa;
  width: 70vw;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:1024px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
/>
`;

exports[`StyledContainer should be rendered 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
/>
`;

exports[`StyledFormContainer should be rendered 1`] = `
.c0 {
  width: 30vw;
  height: 100%;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px 68px;
}

@media (max-width:1024px) {
  .c0 {
    width: 100vw;
    padding: 0px 30px;
  }
}

<div
  className="c0"
/>
`;
